doc/Native APIs.md

```diff
@@ -0,0 +1,417 @@
+- `submission`: The value of the choice that the user selected (string).
+- `message`: General feedback. May contain HTML (string).
+- `tips`: HTML representation of tips. May be `null` (string).
+
+### POST Submit Data
+
+When submitting the problem the data should equal to the string value of the
+selected choice.
+
+Rating Question (`pb-rating`)
+-----------------------------
+
+### `student_view_data`
+
+Identical to MCQ questions except that the `type` field always equals
+`"pb-rating"`.
+
+### `student_view_user_state`
+
+- `student_choice`: The value of the last submitted choice (string).
+
+### `student_results`
+
+Identical to MCQ questions.
+
+### POST Submit Data
+
+When submitting the problem the data should equal to the string value of the
+selected choice.
+
+Multiple Response Question (`pb-mrq`)
+-------------------------------------
+
+### `student_view_data`
+
+- `type`: Always equals `"pb-mrq"` for Multiple Response Question components
+  (string).
+- `id`: Unique ID (name) of the component (string).
+- `title`: Display name of the component (string).
+- `weight`: Weight of the problem (float).
+- `question`: The content of the question (string).
+- `message`: General feedback provided when submitting (string).
+- `hide_results`: `true` if results should be hidden (boolean).
+
+### `student_view_user_state`
+
+- `student_choices`: A list of string values corresponding to choices last
+  submitted by the student (array).
+
+### `student_results`
+
+- `status`: One of `"correct"`, `"incorrect"`, or `"partial"` (string).
+- `score`: Student's score. Float in the range `0 - 1` (float).
+- `weight`: Child component's weight attribute (float).
+- `submissions`: A list of string values corresponding to the choices that the
+  user selected (array).
+- `message`: General feedback. May contain HTML (string).
+- `choices`: A list of dicts containing info about available choices. See below
+  for more info (array).
+
+#### `choices`
+
+Each item in the `choices` array contains these fields:
+
+- `completed`: Boolean indicating whether the state of the choice is correct
+  (boolean).
+- `selected`: `true` if the user selected this choice (boolean).
+- `tips`: Tips formatted as a string of HTML (string).
+- `value`: The value of the choice (string).
+
+### POST Submit Data
+
+The submitted data should be a JSON array of string values corresponding to
+selected choices.
+
+Ranged Value Slider (`pb-slider`)
+---------------------------------
+
+### `student_view_data`
+
+- `type`: Always equals `"pb-slider"` for Ranged Value Slider components (string).
+- `id`: Unique ID (name) of the component (string).
+- `title`: Display name of the component (string).
+- `hide_header`: `true` if header should be hidden (boolean).
+- `question`: The content of the question (string).
+- `min_label`: Label for the low end of the range (string).
+- `max_label`: Label for the high end of the range (string).
```

Comment:
I removed `initial_value` in this PR because it depends on user state. You might have to refresh your cache.